src/test/java/teammates/test/cases/datatransfer/AccountAttributesTest.java

```diff
@@ -133,6 +135,72 @@ public void testLegacyAccountEntityToAttributes() {
 
     }
 
+    @Test
+    public void shouldCreateDeepCopy() {
+        AccountAttributes account = createInvalidAccountAttributesObject();
+        account.isInstructor = true;
+
+        String profileShortName = "John";
+        String profilePersonalEmail = "person@email.com";
+        String profileInstitute = "profile institute test profile 23";
+        String profileNationality = "nationality g35";
+        String profileGender = "male";
+        String profileMoreInfo = "some test info 221";
+        String profilePictureKey = "picture key 223";
+
+        account.studentProfile = StudentProfileAttributes.builder()
+                .withGoogleId(account.googleId)
+                .withShortName(profileShortName)
+                .withEmail(profilePersonalEmail)
+                .withInstitute(profileInstitute)
+                .withNationality(profileNationality)
+                .withGender(profileGender)
+                .withMoreInfo(profileMoreInfo)
+                .withPictureKey(profilePictureKey)
+                .build();
+
+        AccountAttributes copy = account.getCopy();
+
+        assertThat(account, not(copy));
+        assertTrue(account.isInstructor);
+        assertEquals(account.googleId, copy.googleId);
+        assertEquals(account.name, copy.name);
+        assertEquals(account.institute, copy.institute);
+        assertEquals(account.email, copy.email);
+
+        assertThat(account.studentProfile, not(copy.studentProfile));
+        assertEquals(copy.studentProfile.googleId, account.studentProfile.googleId);
```

Comment:
You are testing the `getCopy()` method in `AccountAttribute` and therefore there is no need to test `copy` method in `StudentProfileAttribute`. We shall leave these test cases to `StudentProfileAttributeTest`

Reply:
What about case when `StudentProfileAttribute` will incorrectly copied?

Reply:
The `getCopy()` method inside `StudentProfileAttribute` is responsible to make a copy of the corresponding attributes rather than `getCopy()` method inside `AccountAttribute`. Of course, u need to make sure that the `getCopy()` called `getCopy()` method in `StudentProfileAttribute ` (maybe in this case we just verify some attributes or even just check whether it is null or not). My point here is that each unit test (ideally) should test its own methods rather than the modules that it depends on (or the test case will be a mess).